openupgrade_scripts/scripts/iap/18.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2025 ForgeFlow S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from openupgradelib import openupgrade
+
+model_renames = [
+    ("iap.account.info", "iap.service"),
+]
+
+table_renames = [
+    ("iap_account_info", "iap_service"),
+]
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.logged_query(
+        env.cr,
+        """UPDATE ir_model
+                SET transient = FALSE
+                WHERE model = 'iap.account.info'""",
+    )
+    openupgrade.rename_models(env.cr, model_renames)
+    openupgrade.rename_tables(env.cr, table_renames)
+    openupgrade.add_columns(env, [("iap.service", "integer_balance", "boolean")])
+    openupgrade.logged_query(
+        env.cr,
+        """UPDATE iap_account ia
+        SET name = service_name
+        WHERE COALESCE(name, '') != ''""",
+    )
+    openupgrade.logged_query(
```

Comment:
this needs to go to post-migration

Reply:
https://github.com/OCA/OpenUpgrade/actions/runs/14910707744/job/41884075008?pr=5001#step:13:666